neps/nep-0518.md

```diff
@@ -0,0 +1,337 @@
+
+- **Power and Opacity of Wallet Contract**: The Wallet Contract, pivotal for executing NEAR Actions, may be obscure to Ethereum Wallet users. This necessitates stringent checks and limitations on action parameters, as noted in the proposal to restrict full access key additions.
+
+- **Lack of Batch Transaction Support**: The proposal currently does not accommodate batch transactions available in Wallet Selector, primarily due to limited utility and the complexity it would introduce, along with associated security concerns.
+
+- **[COMPLEX] Potential for inclusion NEAR account functionality in Wallet Contract**: An expanded version of the Wallet Contract, which includes NEAR account functionality, could retain valuable NEAR user patterns. For example, allowing using BOS frontend functionCall keys for Ethereum accounts. This would facilitate simple actions without needing to sign each transaction in the Ethereum wallet. While promising, this approach requires careful management.
+
+- **Wallet Contract Upgrades**: Positioning the Wallet Contract as a protocol feature could limit its upgradability and inadvertently enable smart contract deployments with unbilled storage allocation. Careful implementation can mitigate potential storage growth on NEAR.
+
+- **Exclusion of Native Ethereum Contract Deployment on NEAR**: The proposal does not address the deployment of Ethereum contracts natively on NEAR. Although this is possible via the Aurora network, interactions between NEAR-onboarded Ethereum users and Aurora contracts would necessitate additional steps, like asset bridging or using the [Aurora XCC SDK](https://dev.aurora.dev/posts/cross-ecosystem-communication).
+
+## Future Opportunities
+
+- **Expansion of Ethereum Standards**: A key future development involves simulating other Ethereum standards such as ERC-721, ERC-1155, or ERC-3643. This would enhance NEAR's frontend compatibility with various Ethereum use cases.
+- **Account Abstraction and Extended Features**: This proposal lays the groundwork for account abstraction, potentially leading to the introduction of advanced features in the Wallet Contract. These could include native support for multi-signatures, time-locks, social recovery, and alternative signature algorithms compatible with secure elements in widespread devices.
+- **Aurora Integration**: Exploring integration with Aurora to enable interactions with Aurora-hosted contracts through NEAR, utilizing Ethereum transaction formats and processes.
+- **Bridging and Oracle Onboarding**: This model could facilitate simplified bridging and oracle integration, contributing to the expansion of NEAR's infrastructure.
+
+## Scope of the Project
+
+The scope of this project is extensive, encompassing several critical components:
```

Comment:
Nit: NEP is probably not the best place for project sizing and tracking. One thing to keep in mind is that `nearcore` is just *an* implementation of the NEAR Protocol, and NEPs are meant to describe the changes to *the protocol*, rather than a specific implementation of it.